cmd/kubelet/app/server.go

```diff
@@ -400,6 +407,33 @@ func UnsecuredDependencies(s *options.KubeletServer, featureGate featuregate.Fea
 	if err != nil {
 		return nil, err
 	}
+	var tp oteltrace.TracerProvider
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+	    hostname, err := nodeutil.GetHostname(s.HostnameOverride)
+	    if err != nil {
+		    return nil, err
+	    }
+	    opts := []otlpgrpc.Option{}
+		tc := componentbaseconfigv1alpha1.TracingConfiguration{
+			SamplingRatePerMillion:  s.KubeletConfiguration.Tracing.SamplingRatePerMillion,
+		}
+	    resourceOpts := []otelsdkresource.Option{
+		otelsdkresource.WithAttributes(
+			semconv.ServiceNameKey.String(hostname),
+			semconv.ServiceInstanceIDKey.String(componentKubelet),
+		),
+	    }
+		if s.KubeletConfiguration.Tracing.Endpoint != nil {
```

Comment:
can we add the otelgrpc.WithEndpoint option from within `traces.InitTracing`, and also add `otelgrpc.WithInsecure` there?